src/Recommender/Engagement/FollowButtonTracked.jsx

```diff
@@ -0,0 +1,28 @@
+const handleClick = () => {
+  State.update({ clicked: true });
+};
+
+State.init({
+  clicked: false,
+});
+
+return (
+  <div onClick={handleClick}>
+    <Widget
+      src="${REPL_ACCOUNT}/widget/Recommender.Service.EngagementTracker"
```

Comment:
@roshaans This PR appears to introduce an alternative to the product analytics events recorded in`near-discovery`. Can you help me understand when one should use this approach to record product usage data, versus the approach in `near-discovery`, versus the `data-attr` approach that you shared last week?

Reply:
Hi @charleslavon,

Thank you for bringing this up. Allow me to clarify, please:

**This PR's Approach**: I was specifically tasked to develop a solution focused on click analytics for the recommender components. This method was developed independently of the `data-attr` approach, with no coordination between them.
For testing purposes, I've set up a separate RudderStack account. Additionally, there's also a dedicated source for the recommender features in production. I believe this separation can be beneficial, but I'm open to feedback on this structure.

**data-attr Approach**: I was only recently introduced to this method, last Thursday. Although it's promising, our immediate priority is to deploy engagement tracking for recommendations quickly.

Given the urgency and based on discussions with Pavel, we chose to proceed with the solution in this PR. We're definitely open to consolidating or refining our approach in the future based on feedback and overall strategy.

I hope that clarifies things. Please let me know if you have more questions.

Best,
Daniel

Reply:
thank you for the response @scopalaffairs. 

We currently still do not have a way to track engagement of buttons and components on the BOS yet. Well, until this  [PR 131](https://github.com/NearSocial/VM/pull/131) is merged. 

So till then, we have to use this approach that Daniel worked on.

Reply:
thanks for the detail @scopalaffairs and @roshaans 

Will this data be piped into Pagoda's Rudderstack instance, or is the intent to allow gateway operators to specify a rudderstack destination?

Reply:
@charleslavon currently, it's piped into a specific rudderstack source that shares the prod rudderstack DB deltalake destination with other sources.